### PR TITLE
Foundations

### PR DESCRIPTION
**Pull request by satvikb**
Setup TypeScript, Jest, and file restructures. Basic message handling.

### CODE REVIEW

File: src/messageHandler.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Client, Message } from "discord.js";
+const BOT_ID = "863633669989466142"
+// Refer to https://github.com/discordjs/discord.js/blob/master/typings/index.d.ts for DiscordJS typings
+
+export interface IMessageHandler {
```
Comment (nitinramadoss) [inline src/messageHandler.ts:5]: docs here to explain the interface and subclasses that implement the interface may be helpful

File: src/messageHandler.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  client: Client;
+  receivedMessage:(message : Message) => void;
```
Comment (nitinramadoss) [inline src/messageHandler.ts:7]: do we need more handlers for failed messages?

Comment (satvikb) [reply src/messageHandler.ts:7]: Nope

**Review by nitinramadoss — APPROVED** [review]
LGTM

**Pull request merged**